src/components/DatePicker/DatePicker.tsx

```diff
@@ -0,0 +1,331 @@
+  headerDate.setFullYear(year);
+
+  return (
+    <DatePickerContainer
+      data-testid="datepicker-calendar-container"
+      isResponsive={false}
+      fillWidth={false}
+      orientation="vertical"
+      padding="sm"
+    >
+      <Container
+        isResponsive={false}
+        justifyContent="space-between"
+        orientation="horizontal"
+      >
+        <IconButton
+          icon="chevron-left"
+          onClick={handlePreviousClick}
+          size="sm"
+          type="ghost"
+        />
+        <UnselectableTitle>{headerDateFormatter.format(headerDate)}</UnselectableTitle>
+        <IconButton
+          icon="chevron-right"
+          onClick={handleNextClick}
+          size="sm"
+          type="ghost"
+        />
+      </Container>
+      <DateTable>
+        <thead>
+          <tr>
+            {headers.weekDays.map(({ key, value: date }) => {
+              return (
+                <DateTableHeader key={key}>
+                  {weekdayFormatter.format(date)}
+                </DateTableHeader>
+              );
+            })}
+          </tr>
+        </thead>
+        <tbody>
+          {body.value.map(({ key: weekKey, value: week }) => {
+            return (
+              <tr key={weekKey}>
+                {week.map(({ date, isCurrentMonth, key: dayKey, value: fullDate }) => {
+                  const isSelected = selectedDate
+                    ? isSameDate(selectedDate, fullDate)
+                    : false;
+                  const isCurrentDate = isSameDate(new Date(), fullDate);
+                  const isDisabled = futureDatesDisabled
+                    ? isDateInFuture(fullDate)
+                    : false;
+
+                  return (
+                    <DateTableCell
+                      $isCurrentMonth={isCurrentMonth}
+                      $isDisabled={isDisabled}
+                      $isSelected={isSelected}
+                      $isToday={isCurrentDate}
+                      key={dayKey}
+                      onClick={() => {
+                        if (isDisabled) {
+                          return false;
+                        }
+                        setSelectedDate(fullDate);
+                        closeDatepicker();
+                      }}
+                    >
+                      {date}
+                    </DateTableCell>
+                  );
+                })}
+              </tr>
+            );
+          })}
+        </tbody>
+      </DateTable>
+    </DatePickerContainer>
+  );
+};
+
+export interface DatePickerProps {
+  date?: Date;
+  disabled?: boolean;
+  futureDatesDisabled?: boolean;
+  onSelectDate: (selectedDate: Date) => void;
+  placeholder?: string;
+}
+
+export const DatePicker = ({
+  date,
+  disabled = false,
+  futureDatesDisabled = false,
+  onSelectDate,
+  placeholder,
+}: DatePickerProps) => {
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const [selectedDate, setSelectedDate] = useState<Date>();
```

Comment:
Why keep an internal state instead of just relying on the outside value?

Reply:
Because a default date may not be passed in, but we still need to track the value of the selected date. A default date may be passed in and then a different date selected.